frontend/src/scenes/pipeline/batchExportRunsLogic.tsx

```diff
@@ -0,0 +1,138 @@
+import { lemonToast } from '@posthog/lemon-ui'
+import { actions, afterMount, connect, kea, key, listeners, path, props, reducers, selectors } from 'kea'
+import { loaders } from 'kea-loaders'
+import api, { PaginatedResponse } from 'lib/api'
+import { teamLogic } from 'scenes/teamLogic'
+
+import { BatchExportRun, GroupedBatchExportRuns } from '~/types'
+
+import type { batchExportRunsLogicType } from './batchExportRunsLogicType'
+import { pipelineBatchExportConfigurationLogic } from './pipelineBatchExportConfigurationLogic'
+
+export interface BatchExportRunsLogicProps {
+    id: string
+}
+
+export const batchExportRunsLogic = kea<batchExportRunsLogicType>([
+    props({} as BatchExportRunsLogicProps),
+    key(({ id }) => id),
+    path((key) => ['scenes', 'pipeline', 'batchExportRunsLogic', key]),
+    connect((props: BatchExportRunsLogicProps) => ({
+        values: [
+            teamLogic(),
+            ['currentTeamId'],
+            pipelineBatchExportConfigurationLogic({
+                id: props.id,
+                service: null,
+            }),
+            ['batchExportConfig'],
+        ],
+    })),
+    actions({
+        setDateRange: (from: string | null, to: string | null) => ({ from, to }),
+        switchLatestRuns: (enabled: boolean) => ({ enabled }),
+        loadRuns: true,
+        retryRun: (run: BatchExportRun) => ({ run }),
+    }),
+    loaders(({ props, values }) => ({
+        runsPaginatedResponse: [
+            null as PaginatedResponse<BatchExportRun> | null,
+            {
+                loadRuns: async () => {
+                    // TODO: loading and combining the data could be more efficient
+                    if (values.usingLatestRuns) {
+                        return await api.batchExports.listRuns(props.id, {})
+                    }
+                    return await api.batchExports.listHistoricalRuns(props.id, {
```

Comment:
I'm confused 🤔 why do we need a whole new endpoint for this? Surely it would just be query params given to the existing endpoint?

Reply:
I wrote it this way because return is paged and we need to sort differently - by created_at vs interval_start_at <- see the api. If I used created_at sorting for historical I might miss some runs for the interval, if I use interval_start_at for latest I might not show the runs that were actually latest. Is it possible to parametrise the sorting order?

Reply:
For sure it is yeah, thats what I was thinking - just have it as a query param. I'm not sure if we have a super great. Check Notebooks list UI for an example when you click the different table headers you get different sorting via the `order` param e.g. https://us.posthog.com/api/projects/2/notebooks?order=created_at&limit=50&offset=0